src/api/libcellml/entity.h

```diff
@@ -84,11 +84,37 @@ class LIBCELLML_EXPORT Entity
     bool equals(const EntityPtr &other) const;
 
 protected:
-    Entity(); /**< Constructor, @private. */
     virtual bool doEquals(const EntityPtr &other) const; /**< Virtual implementation method for equals, @private. */
 
+    class EntityImpl; /**< Forward declaration for pImpl idiom, @private. */
+
+    explicit Entity(EntityImpl *derivedPimpl); /**< Constructor for derived classes, @private. */
+
+    /**
+     * @brief Getter for private implementation pointer.
+     *
+     * Getter for private implementation pointer, @private.
+     *
+     * @return A pointer to EntityImpl.
+     */
+    inline EntityImpl *pFunc()
+    {
+        return mPimpl;
+    }
```

Comment:
We are not being consistent:
 - Here, we have a description for the `pFunc()` method, but not in the other headers?
 - Here, we include the body of the `pFunc()` method in the header while it really ought to be in the corresponding `.cpp` file.
 - Some times, we have things in the `protected` section sometimes in the `private` method. I would expect things to be in the `private` method.

Reply:
I need this one to be declared in the header (I think it is Emscripten that won't compile/link if it isn't).